includes/class-wc-subscriptions-cart.php

```diff
@@ -220,7 +220,7 @@ public static function set_subscription_prices_for_calculation( $price, $product
 				if ( $trial_length > 0 ) {
 					$price = $sign_up_fee;
 				} else {
-					$price += $sign_up_fee;
+					$price = (float) $price + $sign_up_fee;
```

Comment:
Without the explicit casting to float, I was getting unit test errors due to price being a string on newer versions of php. This would have been casted automatically in older versions.

Reply:
Just making a note that this seems fine to me. I've noticed WC Core also casts prices to `(float)` with inline comments saying it's for PHP 8 compat.

Reply:
Since that `$price` is the return value, is it worth adding type specifier to this function, e.g. `public static float function set_subscription_prices_for_calculation(...)`? And would that help - i.e. reduce the need for explicit cast?

Reply:
@haszari we still need the explicit as `$price` maybe a string.
I've spent a bit of time playing around with type hinting in php and can cause lots of issues

```php
	public static function set_subscription_prices_for_calculation( float $price, $product ) {
```
Will work find if we send `0` for example. However sending an empty string will throw a `TypeError`. 
Using a straight cast in this instance is the closest to the original intent.